internal/gatewayapi/securitypolicy.go

```diff
@@ -656,10 +702,52 @@ func (t *Translator) translateSecurityPolicyForRoute(
 		}
 
 		irKey := t.getIRKey(gtwCtx.Gateway)
-		for _, listener := range parentRefCtx.listeners {
-			irListener := xdsIR[irKey].GetHTTPListener(irListenerName(listener))
-			if irListener != nil {
+		switch route.GetRouteType() {
```

Comment:
curious why this code is needed when its NA for TCPRoute, which is checked in validateSecurityPolicyForTCP